cmds/boot/boot.go

```diff
@@ -0,0 +1,403 @@
+// Copyright 2012-2017 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// boot allows to handover a system running linuxboot/u-root
+// to a legacy preinstalled operating system by replacing the traditional
+// bootloader path
+
+//
+// Synopsis:
+//	boot
+//
+// Description:
+//	If returns to u-root shell, the code didn't found a local bootable option
+//
+// Notes:
+//	The code is looking for boot/grub/grub.cfg file as to identify the
+//	boot option.
+//	The first bootable device found in the block device tree is the one used
+//	Windows is not supported (that is a work in progress)
+//
+// Example:
+//	boot -v 	- Start the script in verbose mode for debugging purpose
+
+package main
+
+import (
+	"bufio"
+	"encoding/binary"
+	"fmt"
+	"github.com/u-root/u-root/pkg/kexec"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+	"flag"
+	"syscall"
+)
+
+const (
+	bootableMBR = 0xaa55
+	signatureOffset = 510
+)
+
+var verbose bool
+type options struct {
+        verbose bool
+}
+
+func check(e error) {
+	if e != nil {
+		log.Fatal(e)
+	}
+}
+
+func blkDevicesList(blkpath string, devpath string) []string {
+	var blkDevices []string
+	files, err := ioutil.ReadDir(blkpath)
+	if err != nil {
+		log.Fatal(err)
+	}
+	for _, file := range files {
+		deviceEntry, err := ioutil.ReadDir(blkpath + file.Name() + devpath)
+		if err != nil {
+			if ( verbose ) {
+				println("can t read directory")
+			}
+			continue
+		}
+		blkDevices = append(blkDevices, deviceEntry[0].Name())
+	}
+	return (blkDevices)
+}
+
+// checkForBootableMBR is looking for bootable MBR signature 
+// Current support is limited to Hard disk devices and USB devices
+func checkForBootableMBR(path string) (bool,error) {
+	var sig uint16
+	f, err := os.Open(path)
+	if err != nil {
+		return false, err
+	}
+	if err := binary.Read(io.NewSectionReader(f, signatureOffset, 2), binary.LittleEndian, &sig); err != nil {
+		return false,err
+	}
+	if sig != bootableMBR {
+		return false, nil
+	}
+	return true,nil	
+}
+
+// getDevicePartList returns all devices attached to a specific name like /dev/sdaX where X can move from 0 to 127
+// FIXME no support for devices which are included into subdirectory within /dev
+func getDevicePartList(path string) []string {
```

Comment:
Just use filepath.Glob(filepath.Join("/dev", path, "[0-9][0-9]*"))

that will only match your path at least one number.

Reply:
I'll write this and send you the patch if you want.

Reply:
Let me try to do it, I am on the Go learning curve ;). If it doesn't work tonight will ask you for help !